pkg/cliplugins/workload/plugin/workload.go

```diff
@@ -0,0 +1,123 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	kcpclientset "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+)
+
+// Cordon the workload cluster and mark it as unschedulable
+func (c *Config) Cordon(ctx context.Context, workloadClusterName string) error {
+	config, err := clientcmd.NewDefaultClientConfig(*c.startingConfig, c.overrides).ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	err = modifyCordon(ctx, config, workloadClusterName, true)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Uncordon the workload cluster and mark it as schedulable
+func (c *Config) Uncordon(ctx context.Context, workloadClusterName string) error {
+	config, err := clientcmd.NewDefaultClientConfig(*c.startingConfig, c.overrides).ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	err = modifyCordon(ctx, config, workloadClusterName, false)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// change the workload cluster cordon value
+func modifyCordon(ctx context.Context, config *rest.Config, workloadClusterName string, cordon bool) error {
+	kcpClient, err := kcpclientset.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to create kcp client: %w", err)
+	}
+
+	workloadCluster, err := kcpClient.WorkloadV1alpha1().WorkloadClusters().Get(ctx,
+		workloadClusterName,
+		metav1.GetOptions{},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to get workloadcluster %s: %w", workloadClusterName, err)
+	}
+
+	workloadCluster.Spec.Unschedulable = cordon
+	// if uncordon, make sure evictAfter is not set
+	if !cordon {
+		workloadCluster.Spec.EvictAfter = nil
+	}
+
+	_, err = kcpClient.WorkloadV1alpha1().WorkloadClusters().Update(ctx, workloadCluster, metav1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to update workloadcluster %s: %w", workloadClusterName, err)
+	}
+
+	return nil
+}
+
+// Uncordon the workload cluster and mark it as schedulable
+func (c *Config) Drain(ctx context.Context, workloadClusterName string) error {
+	config, err := clientcmd.NewDefaultClientConfig(*c.startingConfig, c.overrides).ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	kcpClient, err := kcpclientset.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to create kcp client: %w", err)
+	}
+
+	workloadCluster, err := kcpClient.WorkloadV1alpha1().WorkloadClusters().Get(ctx,
+		workloadClusterName,
+		metav1.GetOptions{},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to get workloadcluster %s: %w", workloadClusterName, err)
+	}
+
+	nowTime := metav1.NewTime(time.Now())
+	workloadCluster.Spec.EvictAfter = &nowTime
+
+	//ensure unschedulable is also set
+	workloadCluster.Spec.Unschedulable = true
```

Comment:
UX wise, would you want to reject drain if unschedulable is unset, or automatically set it to true?

Reply:
Based on what I see in kubectl, `drain` causes unschedulable to be set and `evictAfter` to be set